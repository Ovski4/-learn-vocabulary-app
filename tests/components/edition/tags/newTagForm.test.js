import reducer from '../../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import NewTagForm from '../../../../src/components/edition/tags/newTagForm/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('NewTagForm component', () => {

    const getStore = (locale) => {
        return createStore(reducer, {
            tags: [],
            config: {
                locale: locale
            }
        });
    }

    it('Should render the component with the english locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('en')}>
                <NewTagForm />
            </Provider>
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });

    it('Should render the component with the french locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('fr')}>
                <NewTagForm />
            </Provider>
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });
});
