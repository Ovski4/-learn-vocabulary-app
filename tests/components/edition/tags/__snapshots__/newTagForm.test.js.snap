// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTagForm component Should render the component with the english locale 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Add a tag
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 5,
        "marginBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Label
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`NewTagForm component Should render the component with the french locale 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "margin": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Créer une étiquette
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "margin": 5,
        "marginBottom": 0,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 5,
          }
        }
      >
        <View
          style={
            Object {
              "minWidth": 50,
            }
          }
        >
          <Text>
            Label
            :
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "flex": 1,
              "marginLeft": 5,
              "paddingLeft": 5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            defaultValue=""
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Object {
                "fontSize": 16,
                "height": 32,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "marginLeft": 5,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={
          Array [
            "disabled",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Valider
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
